tests/integration/test_merge_tree_s3/test.py

```diff
@@ -54,6 +54,13 @@ def cluster():
         logging.info("Cluster started")
         run_s3_mocks(cluster)
 
+        for _, node in cluster.instances.items():
```

Comment:
Why do we need this change now? I mean moving config from main_configs to here.

Reply:
Previously, the instances could still start even if a disk connects to S3 which is not ready yet. Because during initialization, it didn't initialize the disks. They were only initialized at the first call, which was after initializing.

In this PR, because the database_disk is one of the disks under `storage_configuration`, it is used during initialization. So all the disks will be initialized also. If a disk needs to connect to S3, and the S3 has not started yet, the instance cannot start.

To solve it, I remove the `storage_conf.xml` first, so the instances can start. After the S3 started, I added `storage_conf.xml` and  restart the instances.

Reply:
Yes, we can do it in a separate PR if needed. This PR is quite big already.